packages/dd-trace/src/appsec/waf/index.js

```diff
@@ -34,14 +46,43 @@ function destroy () {
   waf.disposeContext = noop
 }
 
-function update (newRules) {
-  // TODO: check race conditions between Appsec enable/disable and WAF updates, the whole RC state management in general
+function checkAsmDdFallback () {
+  if (!waf.wafManager) throw new Error('Cannot update disabled WAF')
+
+  try {
+    waf.wafManager.setAsmDdFallbackConfig()
+  } catch {
+    log.error('[ASM] Could not apply default ruleset back as fallback')
+  }
+}
+
+function updateConfig (product, configId, configPath, config) {
+  if (!waf.wafManager) throw new Error('Cannot update disabled WAF')
+
+  try {
+    if (product === 'ASM_DD') {
+      waf.wafManager.removeConfig(waf.wafManager.constructor.defaultWafConfigPath)
+    }
```

Comment:
if it was me this would be in rule_manager.js, just like it was before this PR, and to do so, you'd need to extract defaultWafConfigPath also in rule_manager, which would pass the default path to the waf constructor etc
but this PR has been going on for too long, so i won't block it